pkg/core/dao/dao.go

```diff
@@ -355,15 +355,22 @@ func (dao *Simple) DeleteStorageItem(scripthash util.Uint160, key []byte) error
 	return dao.Store.Delete(makeStorageItemKey(scripthash, key))
 }
 
-// GetStorageItems returns all storage items for a given scripthash.
-func (dao *Simple) GetStorageItems(hash util.Uint160) (map[string]*state.StorageItem, error) {
-	return dao.GetStorageItemsWithPrefix(hash, nil)
+// StorageItemWithKey is a Key-Value pair together with possible const modifier.
+type StorageItemWithKey struct {
+	state.StorageItem
+	Key []byte
+}
+
+// GetStorageItems returns all storage items for a given scripthash. `needSort` flag
+// specifies whither result items are sorted by keys.
+func (dao *Simple) GetStorageItems(hash util.Uint160, needSort bool) ([]StorageItemWithKey, error) {
+	return dao.GetStorageItemsWithPrefix(hash, nil, needSort)
 }
 
 // GetStorageItemsWithPrefix returns all storage items with given prefix for a
-// given scripthash.
-func (dao *Simple) GetStorageItemsWithPrefix(hash util.Uint160, prefix []byte) (map[string]*state.StorageItem, error) {
-	var siMap = make(map[string]*state.StorageItem)
+// given scripthash. `needSort` flag specifies whither result items are sorted by keys.
+func (dao *Simple) GetStorageItemsWithPrefix(hash util.Uint160, prefix []byte, needSort bool) ([]StorageItemWithKey, error) {
```

Comment:
What's the value of having `needSort` flag? It makes interface a bit overloaded IMO. We can do well with only `GetStorageItems(hash, prefix)` and sorting can be done in `storage.Find` if performance is a bottleneck.